File: math/50. Pow(x, n)/myPow.py
class Solution:
    def myPow(self, x: float, n: int) -> float:
        
        def pow(x, n):
            r = 1
            for i in range(1, n+1):
                r *= x
            return r
        
        if n > 0:
            return pow(x, n)
        elif n < 0:
            if x == 0:
                return 0
            else:
                return pow(1/x, abs(n))
        if n == 0:
            return 1

File: math/50. Pow(x, n)/myPow.js
class Solution {
    myPow(x, n) {
        function pow(x, n) {
            if (x === 0) return 0;
            if (n === 0) return 1;
            
            let res = pow(x, Math.floor(n / 2));
            res *= res;
            
            return n % 2 !== 0 ? x * res : res;
        }
        
        return n >= 0 ? pow(x, n) : 1 / pow(x, Math.abs(n));
    }
}
